design-proposals/app-orch-deploy-applications.md

```diff
@@ -0,0 +1,209 @@
+will need to only create one object in the common case, instead of three.
+
+## Rationale
+
+One alternative approach would be eliminating `Deployment Package` entirely and make
+`Application` a hierarchical object (i.e. An application could contain child Applications).
+This approach has a certain elegance in that it mirrors the Helm pattern where a
+Helm Chart may contain child charts. However, this is seen as too much of a divergence from
+the current implementation as it would require completely refactoring the approach to
+packages. Nothing in this proposal is incompatible with this alternative -- it could be
+taken up as a second phase at a later time.
+
+We could also consider eliminating the `Deployment Package` entirely. If a user wants to
+deploy multiple charts at the same time, we could require them to use an "umbrella helm
+chart", and we would represent the umbrella chart as a single Application object. Eliminating
+objects is disruptive from a backward compatibility / upgrade standpoint. It also drops
+a capability that is currently in use with demonstrated value. We propose re-evaluating
+the prevelance of deployment packages after deployable Applications have been supported
+for a few release cycles, and then making a decision whether to deprecate them.
+
+Another proposal is to eliminate not just `Deployment Package`, but to eliminate `Application`
+as well, and simply put OCI URLs into the `Deployment` objects. This has negative implications
+on reusability -- if the user wishes to create a similar deployment on a second edge, then
+they must re-create the deployment, entering all information a second time. Useful features
+like Profiles and Parameter Templates are lost.
+
+## Affected components and Teams
+
+- Application Orchestration: Catalog and Deployment Services.
+
+- GUI: User Interface.
+
+## Implementation plan
+
+- App Orch adds the new fields to the Application Object, updating API as
+  necessary.
+
+- App Orch modifies the Deployment object and Controller to support deploying applications
+  without deployment packages.
+
+- GUI team modofies the following pages:
+
+  - Update the `Create Deployment` page with a choice to choose between deploying DPs or
+    applications, and proceed with the appropriate workflow.
+
+  - Update the deployment workflow so that in the case of deploying an `Application`, it can use
+    the application profiles directly rather than requiring deployment package profiles.
+
+  - Update the `Edit Deployment` page similar to the changes to `Create Deployment`
+  - Update the `Deployments` list page to support linkage to both Apps and DP
+
+  - Update the `Application` page to have a deployment link and display the `is_deployed` field in both list and detail view.
+
+  - Update any status tables and dashboards as necessary to support these changes.
+
+## Open issues (if applicable)
+
+The requirements on GUI team are considerable. An idea that has been floated is to internally
```

Comment:
This can still be option. If we can convey to user that deployment packages are created automatically for them. In any case user needs to know what is deployment package.

Reply:
added a line about favoring explicit rather than hidden.